src/dbus.c

```diff
@@ -17,106 +17,48 @@
-    "                              Currently defined parameters are:"
-    "                * icon: a string with the path to an icon."
-    "           Method used to notify to the user that the snap that"
-    "           they wanted to run is being refreshed, and they have"
-    "           to wait until the process has finished."
-    "       -->"
-    "       <method name=\"ApplicationIsBeingRefreshed\">"
-    "       <arg direction=\"in\" type=\"s\" name=\"application_name\"/>"
-    "       <arg direction=\"in\" type=\"s\" name=\"lock_file\"/>"
-    "       <arg direction=\"in\" type=\"a{sv}\" name=\"extra_parameters\"/>"
-    "        </method>"
-    "       <!--"
-    "            ApplicationRefreshCompleted:"
-    "            @application_name: the name of the application name"
-    "           @extra_parameters: a dictionary with extra optional parameters."
-    "           Method used to notify the daemon that the refresh is completed"
-    "           and the dialog can be closed."
-    "       -->"
-    "       <method name=\"ApplicationRefreshCompleted\">"
-    "       <arg direction=\"in\" type=\"s\" name=\"application_name\"/>"
-    "       <arg direction=\"in\" type=\"a{sv}\" name=\"extra_parameters\"/>"
-    "        </method>"
-    "    </interface>"
-    "</node>";
-
-    g_autoptr(GDBusNodeInfo) node_info = g_dbus_node_info_new_for_xml (introspection_xml, error);
-    if (node_info == NULL) {
-        g_prefix_error(error, "Failed to parse D-Bus XML: ");
-        return FALSE;
-    }
 
-    static const GDBusInterfaceVTable notifications_vtable = {
-        handle_notifications_method_call,
-        NULL,
-        NULL,
-    };
-    if (g_dbus_connection_register_object(connection,
-                                          "/io/snapcraft/SnapDesktopIntegration",
-                                          node_info->interfaces[0],
-                                          &notifications_vtable,
-                                          (gpointer) state,
-                                          NULL,
-                                          error) == 0) {
-        g_prefix_error(error, "Failed to register notifications object: ");
-        return FALSE;
-   }
-   return TRUE;
+    SnapDesktopIntegration *skeleton = snap_desktop_integration_skeleton_new();
+    g_signal_connect(skeleton, "handle_application_is_being_refreshed",
+                     G_CALLBACK (dbus_handle_application_is_being_refreshed),
+                     state);
+    g_signal_connect(skeleton, "handle_application_refresh_completed",
+                     G_CALLBACK (dbus_handle_close_application_window),
+                     state);
+    g_dbus_interface_skeleton_export (G_DBUS_INTERFACE_SKELETON(skeleton),
+                                      connection,
+                                      "/io/snapcraft/SnapDesktopIntegration",
+                                      NULL);
```

Comment:
This call can fail in the same way that g_dbus_connection_register_object could. Should the error be checked/propagated?

Reply:
Yes. Fixed.